src/components/Slider/Slider.jsx

```diff
@@ -0,0 +1,242 @@
+import React, { forwardRef, useRef } from "react";
+import PropTypes from "prop-types";
+import { COLORS_BASIC, SIZES_BASIC } from "../../constants";
+import useMergeRefs from "../../hooks/useMergeRefs";
+import "./Slider.scss";
+import { bem } from "./SliderCommons";
+import { ensureValueDefault } from "./SliderHelpers";
+import { SliderProvider } from "./SliderContext";
+import SliderBase from "./SliderBase/SliderBase";
+import SliderInfix from "./SliderInfix";
+import { createBemBlockHelper } from "../../helpers/bem-helper";
+
+const Slider = forwardRef(
+  (
+    {
+      ariaLabel,
+      ariaLabeledBy,
+      className,
+      classNameBase,
+      color,
+      "data-testid": dataTestId,
+      disabled,
+      id,
+      isRange,
+      max,
+      min,
+      onChange,
+      step,
+      showValue,
+      size,
+      value,
+      valueDefault,
+      valueFormatter,
+      valueText,
+      // ------ Additional subcomponents' props
+      indicateSelection,
+      prefix,
+      postfix
+    },
+    ref
+  ) => {
+    const componentRef = useRef(null);
+    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
+    const consumerBem = createBemBlockHelper(classNameBase, { isConsume: true });
+    const providerProps = {
+      ariaLabel,
+      ariaLabeledBy,
+      classNameBase,
+      color,
+      dataTestId,
+      disabled,
+      isRange,
+      max,
+      min,
+      onChange,
+      showValue,
+      size,
+      step,
+      value,
+      valueDefault: ensureValueDefault({ valueDefault, isRange, min, max }),
+      valueFormatter,
+      valueText,
+      infixOptions: {
+        prefix,
+        postfix,
+        indicateSelection
+      }
+    };
+    return (
+      <SliderProvider {...providerProps}>
+        <div
+          className={bem("", { disabled, "value-shown": showValue }, className)}
+          data-testid={dataTestId}
+          id={id}
+          ref={mergedRef}
+        >
+          <SliderInfix kind="prefix" />
+          <SliderBase className={consumerBem("base")} />
+          <SliderInfix kind="postfix" />
+        </div>
+      </SliderProvider>
+    );
+  }
+);
+
+Slider.sizes = SIZES_BASIC;
+Slider.colors = COLORS_BASIC;
+
+Slider.propTypes = {
+  // ------ SliderBase props
+  /**
+   * Define a string that labels the current element (Slider)
+   */
+  ariaLabel: PropTypes.string,
+  /**
+   * ElementId of Node that have the text needed for labeling the current element (Slider)
+   */
+  ariaLabeledBy: PropTypes.string,
+  /**
+   * Custom `class name` to be added to the component-root-node
+   */
+  className: PropTypes.string,
+  /**
+   * Custom base of class names for add to component-root and internal-components nodes
+   *  - check Dev Tools for found available Custom/Consumer Classes
+   */
+  classNameBase: PropTypes.string,
```

Comment:
as we talk we will remove this because we want to convert all our components to css modules.

Reply:
Fixed according to comment. 
(https://github.com/mondaycom/monday-ui-react-core/pull/424/commits/1031367619056ae9a182bb88a707aaeefd3c3f04)